File: 0455-assign-cookies/0455-assign-cookies.java
class Solution {
    public int findContentChildren(int[] g, int[] s) {
        int m = g.length;
        int n = s.length;
        //1. Sort the array
        Arrays.sort(g);
        Arrays.sort(s);
        
        //2. Two pointer
        int i = 0, j = 0;
        while(i < m && j < n){
            if(s[j] >= g[i]){
                i++;
            }
            j++;
        }
        return i;
    }
}

File: 0455-assign-cookies/0455-assign-cookies.js
/**
 * @param {number[]} g
 * @param {number[]} s
 * @return {number}
 */
var findContentChildren = function (g, s) {
    g.sort((a, b) => a - b)
    s.sort((a, b) => a - b)
    let i = 0;  //g
    let j = 0;  //s
    let count = 0;

    while (i < g.length && j < s.length) {
        if (s[j] >= g[i]) {
            i++;
            j++;
            count++;
        } else {
            j++;
        }
    }
    return count;
};

// [10,9,8,7] => [7,8,9,10] g
// [5,6,7,8] = > [5,6,7,8] s